src/components/accessories/patientOPD/patientOPD.tsx

```diff
@@ -1,137 +1,158 @@
-    getOpds(patient?.code);
-  }, [patient?.code, shouldUpdateTable, getOpds]);
-
   const userId = useSelector(
     (state: IState) => state.main.authentication.data?.displayName
   );
 
   const diseasesData = useSelector(
     (state: IState) => state.diseases.diseasesOpd.data
   );
+
   useEffect(() => {
     if (activityTransitionState === "TO_RESET") {
-      createOpdReset();
-      setShouldResetForm(true);
       setShouldUpdateTable(true);
+      setCreationMode(true);
+      dispatch(createOpdReset());
+      dispatch(updateOpdReset());
+      setShouldResetForm(true);
     }
-  }, [activityTransitionState, createOpdReset]);
+  }, [dispatch, activityTransitionState, createOpdReset, updateOpdReset]);
 
-  const onSubmit = (createOpdValues: OpdDTO) => {
+  const onSubmit = (opdValuestoSave: OpdDTO) => {
     setShouldResetForm(false);
-    createOpdValues.patientCode = patient?.code;
-    createOpdValues.age = patient?.age;
-    createOpdValues.sex = patient?.sex;
-    createOpdValues.userID = userId;
-    createOpd(createOpdValues, diseasesData);
+    opdValuestoSave.patientCode = patient?.code;
+    opdValuestoSave.age = patient?.age;
+    opdValuestoSave.sex = patient?.sex;
+    opdValuestoSave.userID = userId;
+    if (!creationMode && opdToEdit.code) {
+      dispatch(updateOpd(opdToEdit.code, opdValuestoSave, diseasesData));
+    } else dispatch(createOpd(opdValuestoSave, diseasesData));
   };
 
   const resetFormCallback = () => {
     setShouldResetForm(false);
-    setShouldUpdateTable(false);
+    setCreationMode(true);
+    dispatch(createOpdReset());
+    dispatch(updateOpdReset());
     setActivityTransitionState("IDLE");
+    setShouldUpdateTable(false);
     scrollToElement(null);
   };
+
+  const onEdit = (row: OpdDTO) => {
+    setOpdToEdit(row);
+    setCreationMode(false);
+    scrollToElement(null);
+  };
+
   return (
     <div className="patientOpd">
       <PatientOPDForm
-        fields={initialFields}
+        fields={
+          creationMode
+            ? initialFields
+            : updateOpdFields(initialFields, opdToEdit)
+        }
         onSubmit={onSubmit}
-        submitButtonLabel={t("common.saveopd")}
+        submitButtonLabel={creationMode ? t("opd.saveopd") : t("opd.updateopd")}
         resetButtonLabel={t("common.discard")}
-        isLoading={isLoading}
+        isLoading={createStatus === "LOADING"}
         shouldResetForm={shouldResetForm}
         resetFormCallback={resetFormCallback}
       />
       <div ref={infoBoxRef}>
-        {hasFailed && (
-          <InfoBox
-            type="error"
-            message="Something went wrong, please retry later."
-          />
+        {createStatus === "FAIL" && (
+          <InfoBox type="error" message={t("common.somethingwrong")} />
         )}
       </div>
+      <div ref={infoBoxRef}>
+        {updateStatus === "FAIL" && (
+          <InfoBox type="error" message={t("common.somethingwrong")} />
+        )}
+      </div>
```

Comment:
`<div ref={infoBoxRef}>` should be inside the if statement.

Reply:
Ok, I apply it right now.